src/status_im2/subs/chat/messages.cljs

```diff
@@ -267,3 +268,13 @@
                           joined
                           loading-messages?)
            (albumize-messages))))))
+
+(re-frame/reg-sub
+ :messages/resolve-mention
+ (fn [[_ identity] _]
+   (if (= identity constants/everyone-mention-id)
+     [(atom [(i18n/label :t/everyone-mention)])]
```

Comment:
This looks a bit odd, basically, I don't think we should make a decision in the render and call a different subscription for mentions that are not public keys, so I like the fact that there's a subscription that encapsulate all the logic.
It's though a bit odd that I am shaping the data into what the computation function wants, not sure whether it should be just a signal function and remove the computation function (it's not collating multiple signals anyway), or keeping it as it is.

any opinion?  cc @flexsurfer @ilmotta @erikseppanen @Parveshdhull @ulisesmac 
once we made a decision, I can write a test for it.

Reply:
It looks good that the subcription handles all the logic :+1: 

I'm not sure about "faking" a node in the signal graph (`[(atom [(i18n/label :t/everyone-mention)])]`). IDK if this could break re-frame at any level now or in the future :thinking: , I'd prefer to avoid it  because this could encourage to replicate the pattern elsewhere.

I thought of two alternatives:

1. Constantly (and not dinamically) subscribe and make the desision in the computing function:
```clojure
(re-frame/reg-sub
 :messages/resolve-mention
 (fn [[_ identity] _]
   [(re-frame/subscribe [:contacts/contact-name-by-identity identity])]) ; sub also changed
 (fn [[contact-name] [_ identity]]
   (if (= identity constants/everyone-mention-id)
     (i18n/label :t/everyone-mention)
     contact-name)))
```

2. If we care about subscribing to a value that will probably not be used (such as my first alternative), we could subscribe to nodes closer to the root in the signal graph:

```clojure
(re-frame/reg-sub
 :messages/resolve-mention
 :<- [:contacts/contacts]
 :<- [:multiaccount]
 (fn [[contacts current-multiaccount] [_ identity]]
   (if (= identity constants/everyone-mention-id)
     (i18n/label :t/everyone-mention)
     (-> contacts
         (multiaccounts/contact-by-identity identity)
         (multiaccounts/contact-two-names-by-identity current-multiaccount identity)
         (first)))))
```


What do you think?

Reply:
I like your first suggestion @ulisesmac too.

Reply:
why don't just add a contact for `everyone-mention` id ?

Reply:
we could even create it in status-go, so desktop could use it as well

Reply:
> why don't just add a contact for `constants/everyone-mention-id` ?

I think `contacts` are already enough messy without having also to add synthetic ones :) as well the fact is that eventually we might want to add mention groups (i.e `@mobile`), and sticking everything into `contacts` is likely to become a bit confusing (and as we discussed, contacts are already incredibly confusing).

I like either of @ulisesmac solutions, the only concern is that from what I gathered (tested very quickly), they are a bit more expensive, since we execute the code for `[:contacts/contact-name-by-identity]`, or `:contacts/contacts, :multiaccount`, and `contacts/contacts` is heavy (we should try not to use it).
Generally that would not be an issue, but with messages, it's a place where it's easy to die of a thousand paper cuts, since it's by far the heaviest screen, so not sure, a bit on the fences :( , maybe it doesn't matter so much, but I am conflicted :)

Reply:
>I like either of @ulisesmac solutions, the only concern is that from what I gathered (tested very quickly), they are a bit more expensive

That's the reason I like his first suggestion, but I think the second one causes too much unnecessary recomputation, I'd avoid that.

Reply:
AFAIK, subscriptions are cached and the second alternative is subscribing to the same root nodes as `:contacts/contact-name-by-identity]` & `[:contacts/contact-two-names...]`.

I'm not completely sure, but I'd say that the condition to recompute both the first and second alternatives are the same (they are recomputed if either `:contacts/contacts` or `:multiaccount` change), so probably the performance is almost the same 🤔

I'd implement the solution that I think is better and if a significant performance problem arises, then I'd open an issue to optimize the char screen. Probably there are more fragments of code than this subscription that are computationally expensive in that screen.

Reply:
> AFAIK, subscriptions are cached and the second alternative is subscribing to the same root nodes as :contacts/contact-name-by-identity] & [:contacts/contact-two-names...].

Yes, in the second example `contacts/contacts` would be calculated only once, so you are probably right, I'll double check that, thanks
I think the first one since it's parameterized by `identity`, will trigger `(re-frame/subscribe [:contacts/contact-name-by-identity identity])` for the `@everyone` mention as well, while the second one as you mention will be caching `contacts/contacts`, and always do the work on `displayed-name` for contacts when any of the contact change, while it will short-circuit for mention.

Regardless, happy to go with either of the two,  I think your advice is sound, we can revisit as we go along.
Thanks all for the input!

Reply:
I'm not 100% sure either about how the cache would behave. Only proper measurement can really tell what re-frame is actually doing for this particular sub graph, although I prefer to err on the side of performance.

But also in terms of simplicity. The second option is sort of out of question to me because it's unbundled copy & pasta from the original subscription implementation `:contacts/contact-name-by-identity`. Given that layer-3 subs should always be tested in status-mobile, I think this the second solution doesn't have enough upsides to justify the increased effort + complexity.

If we believe (ideally measure) performance on the same ballpark, then we should pick the simplest.

Reply:
first one it is then :)